pkg/zson/docs/zson-over-json.md

```diff
@@ -0,0 +1,387 @@
+0:[conn;1425565514.419939;CogZFI3py5JsFZGik;fe80::eef4:bbff:fe51:89ec;5353;ff02::fb;5353;udp;dns;15.007272;2148;0;S0;F;F;0;D;14;2820;0;0;[]]
+#1:record[_path:string,ts:time,uid:string,id.orig_h:addr,id.orig_p:port,id.resp_h:addr,id.resp_p:port,proto:enum,trans_id:count,rtt:interval,query:string,qclass:count,qclass_name:string,qtype:count,qtype_name:string,rcode:count,rcode_name:string,AA:bool,TC:bool,RD:bool,RA:bool,Z:count,answers:vector[string],TTLs:vector[interval],rejected:bool]
+1:[dns;1425565514.419939;CogZFI3py5JsFZGik;fe80::eef4:bbff:fe51:89ec;5353;ff02::fb;5353;udp;0;0.119484;_ipp._tcp.local;1;C_INTERNET;12;PTR;0;NOERROR;T;F;F;F;0;[_workstation._tcp.local;sniffer [ec:f4:bb:51:89:ec]._workstation._tcp.local;][4500.000000;4500.000000;]F;]
+0:[conn;1425565545.440378;CoJHyyAGilEHkRZJf;fe80::eef4:bbff:fe51:89ec;5353;ff02::fb;5353;udp;dns;-;-;-;S0;F;F;0;D;1;93;0;0;[]]
+```
+
+## ZSON over JSON
+
+The output below was produced by running
+```
+zq -f zjson input.zson | jq
+```
+This is pretty-printed with jq, but would otherwise be newline-delimited JSON:
+```
+{
+  "id": 0,
+  "type": [
+    {
+      "name": "_path",
+      "type": "string"
+    },
+    {
+      "name": "ts",
+      "type": "time"
+    },
+    {
+      "name": "uid",
+      "type": "string"
+    },
+    {
+      "name": "id.orig_h",
+      "type": "addr"
+    },
+    {
+      "name": "id.orig_p",
+      "type": "port"
+    },
+    {
+      "name": "id.resp_h",
+      "type": "addr"
+    },
+    {
+      "name": "id.resp_p",
+      "type": "port"
+    },
+    {
+      "name": "proto",
+      "type": "enum"
+    },
+    {
+      "name": "service",
+      "type": "string"
+    },
+    {
+      "name": "duration",
+      "type": "interval"
+    },
+    {
+      "name": "orig_bytes",
+      "type": "count"
+    },
+    {
+      "name": "resp_bytes",
+      "type": "count"
+    },
+    {
+      "name": "conn_state",
+      "type": "string"
+    },
+    {
+      "name": "local_orig",
+      "type": "bool"
+    },
+    {
+      "name": "local_resp",
+      "type": "bool"
+    },
+    {
+      "name": "missed_bytes",
+      "type": "count"
+    },
+    {
+      "name": "history",
+      "type": "string"
+    },
+    {
+      "name": "orig_pkts",
+      "type": "count"
+    },
+    {
+      "name": "orig_ip_bytes",
+      "type": "count"
+    },
+    {
+      "name": "resp_pkts",
+      "type": "count"
+    },
+    {
+      "name": "resp_ip_bytes",
+      "type": "count"
+    },
+    {
+      "name": "tunnel_parents",
+      "type": "set[string]"
+    }
+  ],
+  "value": [
+    "conn",
+    "1425565514.419939",
+    "CogZFI3py5JsFZGik",
+    "fe80::eef4:bbff:fe51:89ec",
+    "5353",
+    "ff02::fb",
+    "5353",
+    "udp",
+    "dns",
+    "15.007272",
+    "2148",
+    "0",
+    "S0",
+    "F",
+    "F",
+    "0",
+    "D",
+    "14",
+    "2820",
+    "0",
+    "0",
+    null
```

Comment:
shouldn't this be `[]` rather than null?

Reply:
(playing around with `zq` a bit, i see this comes out as `-` in the zeek format, so it's probably a pre-existing issue).

Reply:
I thought it would be easier for a pure javascript client to parse so I left it this way.

Reply:
I guess it boils down to whether non-zson formats and their associated codecs should all be "lossless" allowing a round-trip encode/decode to get the original zson back, or if it's ok for them to be one-way. "Lossless" seems preferable to me, but either way it doesn't have to be settled here.

Reply:
It definitely should be lossless.  I thought null would be interpreted by the client as empty set here.  But I see your point here.  [] would be more straightforward for the client.  I'll fix it.